executor/src/witgen/machines/machine_extractor.rs

```diff
@@ -57,25 +57,26 @@ pub fn split_out_machines<'a, T: FieldElement>(
             base_identities.iter().cloned().partition(|i| {
                 // The identity has at least one machine witness, but
                 // all referenced witnesses are machine witnesses.
-                let all_refs = &refs_in_identity(i) & (&all_witnesses);
+                let all_refs = &refs_in_selected_expressions(&i.left) & (&all_witnesses);
                 !all_refs.is_empty() && all_refs.is_subset(&machine_witnesses)
             });
         base_identities = remaining_identities;
         remaining_witnesses = &remaining_witnesses - &machine_witnesses;
 
-        let connecting_identities = base_identities
+        let connecting_identities = identities
```

Comment:
Maybe add a comment: These are the "calling" identities.

Reply:
Actually a bit unrelated to this PR: So we are actually determining all the "calling" identities, but IIRC, we do not automatically query the machine related to the connecting identity, do we, instead we iterate through all machines - or is that fixed already?

Reply:
Yes, that is still the case!